cfme/cloud/tenant.py

```diff
@@ -194,48 +192,93 @@ def is_displayed(self):
 
         all_view = self.create_view(TenantAllView)
-        wait_for(lambda: all_view.is_displayed, fail_condition=False, num_sec=120, delay=3,
-                 fail_func=lambda: all_view.flush_widget_cache(), handle_exception=True)
-        if cancel:
+        wait_for(lambda: all_view.is_displayed, num_sec=120, delay=3,
+                 fail_func=all_view.flush_widget_cache, handle_exception=True)
+
+        if not changed:
             if self.appliance.version >= '5.8':
                 msg = 'Add of Cloud Tenant was cancelled by the user'
             else:
                 msg = 'Add of new Cloud Tenant was cancelled by the user'
             all_view.entities.flash.assert_success_message(msg)
         else:
             all_view.entities.flash.assert_success_message(
-                'Cloud Tenant "{}" created'.format(self.name))
+                'Cloud Tenant "{}" created'.format(name))
+
+        tenant = Tenant(name, provider)
+
         if wait:
             def refresh():
                 """Refresh a few things"""
-                self.provider.refresh_provider_relationships()
+                tenant.provider.refresh_provider_relationships()
                 all_view.flush_widget_cache()
                 self.browser.refresh()
 
-            wait_for(lambda: self.exists, timeout=600, message='Wait for cloud tenant to appear',
+            wait_for(lambda: tenant.exists, timeout=600,
+                     message='Wait for cloud tenant to appear',
                      delay=10, fail_func=refresh)
 
+        return tenant
+
+    def delete(self, *tenants):
+        """Delete one or more  Tenants from the list of the Tenants
+
+        Args:
+            list of the `cfme.cloud.tenant.Tenant` objects
+        """
+
+        tenants = list(tenants)
+        checked_tenants = []
+        view = navigate_to(self, 'All')
+        # double check we're in List View
+        view.toolbar.view_selector.select('List View')
+        if not view.entities.table.is_displayed:
+            raise ValueError('No Tenants found')
+        for row in view.entities.table:
+            for tenant in tenants:
+                if tenant.name == row.name.text:
+                    checked_tenants.append(tenant)
+                    row[0].check()
+                    break
+            if set(tenants) == set(checked_tenants):
+                break
+        if set(tenants) != set(checked_tenants):
+            raise ValueError('Some tenants were not found in the UI')
+        view.toolbar.configuration.item_select('Delete Cloud Tenants', handle_alert=True)
+        for tenant in tenants:
+            tenant.wait_for_disappear()
+        view.entities.flash.assert_no_error()
+
+        # TODO: Assert deletion flash message for selected tenants
+        # it is not shown in current UI, so not asserting
+
+
+class Tenant(Navigatable):
+    '''Tenant Class'''
+    _param_name = 'Tenant'
+
+    def __init__(self, name, provider, appliance=None):
```

Comment:
This should accept the parent collection instead of an appliance, getting the appliance from the collection.